src/EnergyPlus/SetPointManager.cc

```diff
@@ -318,3730 +370,1075 @@ void GetSetPointManagerInputData(EnergyPlusData &state, bool &ErrorsFound)
-        } else if (Util::SameString(state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).ctrlVarType, "MaximumHumidityRatio")) {
-            state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).CtrlTypeMode = CtrlVarType::MaxHumRat;
-        } else if (Util::SameString(state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).ctrlVarType, "MinimumHumidityRatio")) {
-            state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).CtrlTypeMode = CtrlVarType::MinHumRat;
-        } else if (Util::SameString(state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).ctrlVarType, "MassFlowRate")) {
-            state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).CtrlTypeMode = CtrlVarType::MassFlow;
-        } else if (Util::SameString(state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).ctrlVarType, "MaximumMassFlowRate")) {
-            state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).CtrlTypeMode = CtrlVarType::MaxMassFlow;
-        } else if (Util::SameString(state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).ctrlVarType, "MinimumMassFlowRate")) {
-            state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).CtrlTypeMode = CtrlVarType::MinMassFlow;
-        } else {
-            ShowSevereError(state, format("{}: {}=\"{}\", invalid field.", RoutineName, cCurrentModuleObject, cAlphaArgs(1)));
-            ShowContinueError(state, format("..invalid {}=\"{}\".", cAlphaFieldNames(2), cAlphaArgs(2)));
-            ShowContinueError(state, R"(..Valid values are "Temperature","MaximumTemperature","MinimumTemperature",)");
-            ShowContinueError(state, R"(     "HumidityRatio","MaximumHumidityRatio","MinimumHumidityRatio","MassFlowRate",)");
-            ShowContinueError(state, R"(     "MaximumMassFlowRate" or "MinimumMassFlowRate")");
-            ErrorsFound = true;
-        }
-
-        state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).Sched = cAlphaArgs(3);
-        state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).SchedPtr = GetScheduleIndex(state, cAlphaArgs(3));
-        if (state.dataSetPointManager->SchSetPtMgr(SetPtMgrNum).SchedPtr == 0) {
-            if (lAlphaFieldBlanks(3)) {
-                ShowSevereError(state, format("{}: {}=\"{}\", blank required field.", RoutineName, cCurrentModuleObject, cAlphaArgs(1)));
-                ShowContinueError(state, format("..required field {}", cAlphaFieldNames(3)));
+    for (int iSPM = 0; iSPM < (int)SPMType::Num; ++iSPM) {
+        SPMType type = static_cast<SPMType>(iSPM);
+        cCurrentModuleObject = spmTypeNames[iSPM];
+
+        auto const instances = ip->epJSON.find(cCurrentModuleObject);
+        if (instances == ip->epJSON.end()) continue; // No SetPointManagers of this type
+
+        auto const &props = ip->getObjectSchemaProps(state, cCurrentModuleObject);
+
+#define PRESERVE_IDF_ORDER        
+#ifdef PRESERVE_IDF_ORDER
+        
+        // Try to read these in IDF order (for now, can revert to simpler begin(), ++, end() iterator later)
```

Comment:
If you want an example of how to read epJson objects (useful for tracking down diffs), here it is. Same approach is used in `getObjectItem`.